uor.py

```diff
@@ -53,17 +54,17 @@ def main():
 
 	### Init and parsing
 
-	if len(argv) > 1:
-		for arg in argv[1:]:
-			if not arg.startswith('--'):
-				url = arg
-				break
-
 	GUI = not is_terminal()
-	if "--gui" in argv: GUI = True
-	if "--cli" in argv: GUI = False
-	if "--no-gui" in argv: GUI = False
-	if "--no-cli" in argv: GUI = True
+
+	parser = argparse.ArgumentParser()
+	grp = parser.add_mutually_exclusive_group()
+	grp.add_argument('--gui', action='store_true', default=GUI)
+	grp.add_argument('--cli', action='store_false', dest='gui')
```

Comment:
I'm curious why `--cli` would even be an option? Wouldn't it be simpler to get rid of `add_mutually_exclusive_group` and this `--cli` option?

Even the code below essentially does this?

```python
if GUI: ...
else: ...  # no "if CLI:" needed?
```

Reply:
```py3
def is_terminal() -> bool:
    import os
    return os.isatty(0) and os.isatty(1)

def main():
    # Some other lines
    GUI = not is_terminal()

    parser = argparse.ArgumentParser()
    grp = parser.add_mutually_exclusive_group()
    grp.add_argument('--gui', action='store_true', default=GUI)
    grp.add_argument('--cli', action='store_false', dest='gui')
```

`GUI` may be `True` in some cases e.g. either the input or the output is a pipe and the user wants to force it to `False`.

Reply:
I said a pipe, but I meant a file, sorry

Reply:
The idea is auto detect: if you are in a terminal, use the CLI by default. Otherwise, use the GUI one. But if the user wants, it can overwrite the auto-detection. Another way could be `--gui` and `--no-gui`. The initial implementation was to be fast to test, not to be decent. Let's fix that.


I will continue the discussion on #2.